webapp/lib/controller.dart

```diff
@@ -90,6 +102,61 @@ void initUI() {
   );
 }
 
+void setupProjectTimers() {
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+  for (var project in allProjects) {
+    var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+    projectTimers.add(new ProjectTimer(project, timer));
+  }
+}
+
+void setupProjectTimer(String project) {
+  var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+  projectTimers.add(new ProjectTimer(project, timer));
```

Comment:
I think expired timers should be removed from the `projectTimers` list? Otherwise you end up with multiple timers per project...
Having said that, since it's a 1:1 mapping between projects and timers, and there should only be one timer per project at any time, I suggest to just have `projectTimers` as a `Map<String, Timer` rather than a list of project-timer pair objects, which should also make retrieving the timer for each project much simpler.

Reply:
made project timers a map